### PR TITLE
chore(app, api-client): remove mock RTP datafrom ProtocolDetails

### PR DESCRIPTION
**Pull request by ncdiehl11**
closes [AUTH-266](https://opentrons.atlassian.net/browse/AUTH-266)

# Overview

Remove mock RTP data from ProtocolDetails. Land on parameters tab for RTP-containing protocols.

# Test Plan

### RTP protocol
- upload a protocol containing RTPs
- select protocol card to land on protocol details screen
- verify that you land on 'parameters' tab and protocol parameters are mapped properly (read only)

### non-RTP protocol
- upload a protocol _not_ containing RTPs
- select protocol card to land on protocol details screen
- verify that you land on 'hardware' tab
- verify that 'parameters' tab renders 'No parameters specified in this protocol'

https://github.com/Opentrons/opentrons/assets/47604184/c72b45d4-ed9b-4a90-80cd-119c6d068ab0

# Changelog

- remove mock data
- on ProtocolDetails, pull RTPs from `mostRecentAnalysis`
- update protocol test fixture to include RTPs

# Review requests

authorship devs

# Risk assessment

low

[AUTH-266]: https://opentrons.atlassian.net/browse/AUTH-266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api-client/src/protocols/__fixtures__/simpleAnalysisFile.json [hunk 1]
```diff
@@ -3937,5 +3937,16 @@
       "displayColor": "#b925ff"
     }
   ],
-  "runTimeParameters": []
+  "runTimeParameters": [
```
Comment (koji) [inline api-client/src/protocols/__fixtures__/simpleAnalysisFile.json:3940]: can we have at least 4 types as a fixture?
int, float, str, bool

**Review by koji — APPROVED** [review]
lgtm
thanks!

File: api-client/src/protocols/__fixtures__/simpleAnalysisFile.json (hunk 1 above)
Comment (jerader) [reply api-client/src/protocols/__fixtures__/simpleAnalysisFile.json:3940]: ^ I agree, would be nice to have one example of each

**Review by jerader — APPROVED** [review]
can you remove the mock data in `useRunTimeParameters` as well?

File: api-client/src/protocols/__fixtures__/simpleAnalysisFile.json [hunk 2]
```diff
@@ -3937,5 +3937,57 @@
...
+    {
```
Comment (koji) [inline api-client/src/protocols/__fixtures__/simpleAnalysisFile.json:3941]: thanks @ncdiehl11

**Pull request merged**